### PR TITLE
🍰 1054 blocked users

### LINKED ISSUE #1054: 🚀 [Feature] Block User
## :rocket: [Feature] Block User

~~~gherkin
Feature: As a user
I'd like to block another user
to hinder him to see and interact with my contributions, profile and comments
and also that I don't see his profile, contributions and comments any longer.
~~~

### Design & Layout

block-user
blocked-users

### Additional context

This is part of Feature Blacklist #279 and a working out of #222. It should also be possible to unblock a user again later the same way.

### PR DESCRIPTION
**Pull request by roschaefer**
### Issues

- fixes #1054 

### Todo

- [X] None

### CODE REVIEW

File: backend/src/schema/resolvers/users/blockedUsers.spec.js [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+const instance = neode()
+
+let currentUser
+let blockedUser
+let authenticatedUser
+let server
+
+beforeEach(() => {
+  authenticatedUser = undefined
+  ;({ server } = createServer({
```
Comment (Tirokk) [inline backend/src/schema/resolvers/users/blockedUsers.spec.js]: I never saw this notation before.
What does the `;` mean? @roschaefer

File: webapp/locales/de.json [hunk 2]
```diff
@@ -189,6 +189,25 @@
...
+        "intro": "Wenn ein anderer Benutzer von dir blockiert wurde, dann passiert folgendes:",
+        "your-perspective": "In deiner Beitragsübersicht tauchen keine Beiträge der blockierten Person mehr auf.",
+        "their-perspective": "Umgekehrt das gleiche: Die blockierte Person sieht deine Beiträge auch nicht mehr in ihrer Übersicht.",
+        "search": "Die Beiträge von blockierten Personen verschwinden aus deinen Suchergebnissen.",
+        "notifications": "Von dir blockierte Personen erhalten keine Benachrichtigungen mehr, wenn sie in deinen Beiträgen erwähnt werden.",
+        "closing": "Das sollte fürs Erste genügen, damit blockierte Benutzer dich nicht mehr länger belästigen können."
+      },
+      "columns": {
+        "name": "Name",
+        "slug": "Alias"
```
Comment (Tirokk) [inline webapp/locales/de.json]: Why here `Alias` and in English `Slug`? @roschaefer 

In the Admins Users list in both languages the slug is name `Slug`.
`Alias`is better understandable for the users for sure!

File: webapp/locales/en.json [hunk 3]
```diff
@@ -189,6 +189,26 @@
...
+        "your-perspective": "The blocked person's posts will no longer appear in your news feed.",
+        "their-perspective": "Vice versa: The blocked person will also no longer see your posts in their news feed.",
+        "search": "Posts of blocked people disappear from your search results.",
+        "notifications": "Blocked users will no longer receive notifications if they are mentioned in your posts.",
+        "closing": "This should be sufficient for now so that blocked users can no longer bother you."
+
+      },
+      "columns": {
+        "name": "Name",
+        "slug": "Slug"
```
Comment (Tirokk) [inline webapp/locales/en.json]: See above …

File: webapp/locales/de.json (hunk 2 above)
Comment (roschaefer) [reply webapp/locales/de.json]: I thought that "Slug" is not so common in German for most people to understand. I don't know what it actually translates to, I believe it's "Schnecke" and refers to the pattern of strings that have no witespaces as delimiter (in a URL).

Good luck understanding that one..

File: backend/src/schema/resolvers/users/blockedUsers.spec.js (hunk 1 above)
Comment (roschaefer) [reply backend/src/schema/resolvers/users/blockedUsers.spec.js]: It was added by eslint. The problem is that if you have e.g.
```js
const { a } = something
```
is valid and no problem but
```js
let a
{ a } = something
```
then babel gets confused. You need to wrap it with parantheses and apparently it's good practice (?) to add a semicolon as delimiter?

**Comment by roschaefer** [comment]
@Tirokk forgot to add 31299235a3850e47ed885cf71579c3117da52dac could you recheck, please?

File: backend/src/schema/resolvers/users/blockedUsers.spec.js (hunk 1 above)
Comment (Tirokk) [reply backend/src/schema/resolvers/users/blockedUsers.spec.js]: Ah, okay.

File: webapp/locales/de.json (hunk 2 above)
Comment (Tirokk) [reply webapp/locales/de.json]: But does an English user understands `Slug` better as a German? @roschaefer

Comment (roschaefer) [reply webapp/locales/de.json]: No, I think "Slug" is also a bad name. Shall we use "alias" everywhere?

**Comment by Tirokk** [comment]
DE: Nun funzt es. Wir fliegen zu den Sternen …
EN: Now it’s functioning. We flying to the stars …

Cool thing @roschaefer !!! ❄️💫🚀🚀🚀

Wisdom

**Comment by Tirokk** [comment]
Love to see what @mattwr18 says about the `Slug` …

File: webapp/locales/de.json (hunk 2 above)
Comment (mattwr18) [reply webapp/locales/de.json]: I understand what a slug is, but can't remember when I gained that understanding :woman_shrugging: 
An alias, I guess is better understood by the general user, but it's not really the same, is it?

**Pull request merged**